inc/ThirdParty/Hostings/WPEngine.php

```diff
@@ -0,0 +1,160 @@
+<?php
+
+namespace WP_Rocket\ThirdParty\Hostings;
+
+use WP_Rocket\ThirdParty\ReturnTypesTrait;
+use WP_Rocket\Event_Management\Subscriber_Interface;
+use WpeCommon;
+
+/**
+ * Compatibility class for WP Engine.
+ *
+ * @since 3.6.1
+ */
+class WPEngine implements Subscriber_Interface {
+	use ReturnTypesTrait;
+
+	/**
+	 * Array of events this subscriber wants to listen to.
+	 *
+	 * @since 3.6.1
+	 *
+	 * @return array
+	 */
+	public static function get_subscribed_events() {
+		if (
+		! (
+			class_exists( 'WpeCommon' )
+			&&
+			function_exists( 'wpe_param' )
+		)
+		) {
+			return [];
+		}
+
+		return [
+			'rocket_varnish_field_settings'           => 'varnish_addon_title',
+			'rocket_display_input_varnish_auto_purge' => 'return_false',
+			'rocket_cache_mandatory_cookies'          => [ 'return_empty_array', PHP_INT_MAX ],
+			'rocket_advanced_cache_file'              => 'return_empty_string',
+			'admin_init'                              => [
+				[ 'remove_notices' ],
+				[ 'run_rocket_bot_after_wpengine' ],
+			],
+			'set_rocket_wp_cache_define'              => 'return_true',
+			'do_rocket_generate_caching_files'        => 'return_false',
+			'after_rocket_clean_domain'               => 'clean_wpengine',
+			'rocket_buffer'                           => [ 'add_footprint', 50 ],
+			'rocket_disable_htaccess'                 => 'disable_htaccess',
+		];
+	}
+
+	/**
+	 * Changes the text on the Varnish one-click block.
+	 *
+	 * @since 3.6.1
+	 *
+	 * @param array $settings Field settings data.
+	 *
+	 * @return array modified field settings data.
+	 */
+	public function varnish_addon_title( $settings ) {
+		$settings['varnish_auto_purge']['title'] = sprintf(
+			// Translators: %s = Hosting name.
+			__( 'Your site is hosted on %s, we have enabled Varnish auto-purge for compatibility.', 'rocket' ),
+			'WP Engine'
+		);
+
+		return $settings;
+	}
+
+	/**
+	 * Stop showing not valid notices with WP Engine.
+	 *
+	 * @since 3.6.1
+	 */
+	public function remove_notices() {
+		$container  = apply_filters( 'rocket_container', null );
+		$subscriber = $container->get( 'admin_cache_subscriber' );
+
+		remove_action( 'admin_notices', [ $subscriber, 'notice_advanced_cache_permissions' ] );
```

Comment:
@engahmeds3ed Can you check the processing order of when the `'notice_advanced_cache_permissions'` callback is registered to `'admin_notices'`? 

Why?

`remove_action` must happen _after_ the `add_action`

Reply:
Ok will check but I think this block of code was adjusted here #2747

Reply:
I checked that and found that `remove_action` is already called after `add_action` as follows:-
- add_action
https://github.com/wp-media/wp-rocket/blob/0416aff5240e0cf873d1f54b8b3cb7a86186b446/inc/Engine/Cache/AdminSubscriber.php#L46
https://github.com/wp-media/wp-rocket/blob/1abe50e6a004927a0aeb93879a9c1352d4f8bcd3/inc/Engine/Cache/ServiceProvider.php#L45
https://github.com/wp-media/wp-rocket/blob/da2381794a10862aaf9159e444e1c2ef321af1fe/inc/Plugin.php#L104

- remove_action
https://github.com/wp-media/wp-rocket/blob/5633f3ec0275b6f1e59f73ac7df109c578b69ea1/inc/ThirdParty/Hostings/WPEngine.php#L80
https://github.com/wp-media/wp-rocket/blob/da2381794a10862aaf9159e444e1c2ef321af1fe/inc/Plugin.php#L257